src/target/riscv/riscv-013.c

```diff
@@ -3424,114 +3412,189 @@ static void log_mem_access_result(struct target *target, bool success, int metho
 		LOG_TARGET_DEBUG(target, "%s", msg);
 }
 
-static bool mem_should_skip_progbuf(struct target *target, target_addr_t address,
-		uint32_t size, bool is_read, char **skip_reason)
+typedef enum {
+	MEM_ACCESS_STATUS_KIND_OK,
+	MEM_ACCESS_STATUS_KIND_DISABLED,
+	MEM_ACCESS_STATUS_KIND_SKIPPED,
+	MEM_ACCESS_STATUS_KIND_FAILED
+} mem_access_status_kind_t;
+
+#define LIST_OF_MEM_ACCESS_STATUSES                                                                \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_OK,                                            \
+			MEM_ACCESS_STATUS_KIND_OK, "ok")                                           \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_DISABLED,                                      \
+			MEM_ACCESS_STATUS_KIND_DISABLED, "disabled")                               \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_SKIPPED_ABSTRACT_ACCESS_CMDERR,                \
+			MEM_ACCESS_STATUS_KIND_SKIPPED, "skipped (abstract access cmderr)")        \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_SKIPPED_PROGBUF_INSUFFICIENT,                  \
+			MEM_ACCESS_STATUS_KIND_SKIPPED, "skipped (insufficient progbuf)")          \
```

Comment:
Here it'd be good to introduce also "no progbuf" status:

```
	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_SKIPPED_PROGBUF_NOT_PRESENT,                  \
			MEM_ACCESS_STATUS_KIND_SKIPPED, "skipped (progbuf not present)")          \
```

This is however an improvement for a separate PR.

Reply:
Addressed

Reply:
Please:

- either keep the new status and use it where suitablem
- or remove it for now. It can be added later in a separate PR as an improvement.

My intention was to give a very clear error message to the user: If there is no program buffer in a given target, it is better to say `no progbuf` than `insufficient progbuf`, since the latter could be misleading. The user could think that a progbuf is actually there but not large enough.

(I am sorry for my not very clear previous comment.)

Reply:
Added usage of a new status

Reply:
Thank you